src/calc.c

```diff
@@ -310,8 +316,23 @@ static char* calc_preprocess_input(Mode* sw, const char* input)
     GError *error = NULL;
     CALCModePrivateData* pd = (CALCModePrivateData*)mode_get_private_data(sw);
 
-    const gchar* const argv[] = { "qalc", "+u8", "-s", "update_exchange_rates 1days", input, NULL };
+    const gchar* const default_args[] = { "+u8", "-s", "update_exchange_rates 1days", input, NULL };
+    int len = sizeof(default_args) / sizeof(const gchar*);
+
+    int n_extra_args = 2; // "qalc" and optional  "-t"
+    const gchar** argv = malloc((len + n_extra_args) * sizeof(const gchar*));
+
+    int i = 0;
+    argv[i++] = "qalc";
+    if (find_arg(TERSE_OPTION) > -1) {
+        argv[i++] = "-t";
+    }
+    for (int j = 0; j < len; j++) {
```

Comment:
The reason I think using `GArray` would be nice here is so you can get read of this for loop as you can just insert directly into the old array.